src/SimpleMemSystem.cpp

```diff
@@ -274,227 +323,65 @@ static void Trace(etiss::uint64 addr, etiss::uint32 len, bool isWrite, bool toFi
         std::cout << text.str();
 }
 
-etiss::int32 SimpleMemSystem::dread(ETISS_CPU *cpu, etiss::uint64 addr, etiss::uint8 *buf, etiss::uint32 len)
-{
-    if (len > 0)
-    {
-        int i_seg = 0;
-        int n_segs = msegs_.size();
-        size_t offset = 0;
-        for (i_seg = 0; i_seg < n_segs; ++i_seg)
-        {
-            if (msegs_[i_seg]->addr_in_range(addr))
-            {
-                offset = addr - msegs_[i_seg]->start_addr_;
-                break;
-            }
-        }
+etiss::int32 SimpleMemSystem::dbus_access(ETISS_CPU *cpu, etiss::uint64 addr, etiss::uint8 *buf, etiss::uint32 len, bool write) {
+    uint64 pc = cpu ? cpu->instructionPointer : 0;
```

Comment:
can cpu be null here? if not, this could be an assert for better performance.

Reply:
Unfortunately it can, cpu is used for debug / error prints.